openapi2conv/openapi2_conv.go

```diff
@@ -638,6 +641,49 @@ func FromV3Swagger(swagger *openapi3.Swagger) (*openapi2.Swagger, error) {
 	return result, nil
 }
 
+func consumesToArray(consumes map[string]bool) []string {
+	var consumesArr []string
+	for key, _ := range consumes {
+		consumesArr = append(consumesArr, key)
+	}
+	sort.Strings(consumesArr)
+	return consumesArr
+}
+
+func FromV3RequestBodies(name string, requestBodyRef *openapi3.RequestBodyRef, components *openapi3.Components) (*openapi2.Parameters, *openapi2.Parameters, map[string]bool, error) {
+	formParameters := openapi2.Parameters{}
+	bodyOrRefParameters := openapi2.Parameters{}
+	if ref := requestBodyRef.Ref; ref != "" {
+		bodyOrRefParameters = append(bodyOrRefParameters, &openapi2.Parameter{Ref: FromV3Ref(ref)})
+		return &bodyOrRefParameters, &formParameters, nil, nil
+	}
+
+	//Only select one formData or request body for an individual requesstBody as swagger 2 does not support multiples
```

Comment:
How about returning an error "both form data and request body were provided"

Reply:
again, is the priority only to convert when it is perfectly backward compatible?

Reply:
You're right: let's do best effort.